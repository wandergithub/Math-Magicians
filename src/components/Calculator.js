import React from 'react';
import Display from './Display';
import Button from './Button';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="calculatorContainer">
        <Display />
        <div className="buttons-container">
          <div className="row">
            <Button value="AC" />
            <Button value="+/-" />
            <Button value="%" />
            <Button value="รท" />
          </div>
          <div className="row">
            <Button value="7" />
            <Button value="8" />
            <Button value="9" />
            <Button value="x" />
          </div>
          <div className="row">
            <Button value="4" />
            <Button value="5" />
            <Button value="6" />
            <Button value="-" />
          </div>
          <div className="row">
            <Button value="1" />
            <Button value="2" />
            <Button value="3" />
            <Button value="+" />
          </div>
          <div className="row row-5">
            <Button value="0" />
            <Button value="." />
            <Button value="=" />
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
