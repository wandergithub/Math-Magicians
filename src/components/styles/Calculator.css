.calculator-container {
  display: flex;
  justify-content: space-around;
  margin-top: 5%;
}

.calculator-container h2 {
  font-size: 2em;
}
