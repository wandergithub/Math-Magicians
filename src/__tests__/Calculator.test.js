import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import {
  render, fireEvent, waitFor, screen,
} from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator component tests', () => {
  it('Renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Calculator />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('User click interaction with calculator buttons', () => {
    render(<Calculator />);
    const display = screen.getByText('00');
    const button = screen.getByText('8')

    fireEvent.click(button);

    expect(display.innerHTML).toBe("8");
  });
});
