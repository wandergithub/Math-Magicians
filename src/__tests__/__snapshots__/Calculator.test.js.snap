// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component tests Renders correctly 1`] = `
<div
  className="calculatorContainer"
>
  <div
    className="outputContainer"
  >
    <p>
      0
      0
    </p>
  </div>
  <div
    className="buttons-container"
  >
    <div
      className="row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row row-5"
    >
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
